internal/bioscfg/action.go

```diff
@@ -0,0 +1,104 @@
+package bioscfg
+
+import (
+	"context"
+	"io"
+	"net/http"
+
+	"github.com/metal-toolbox/bioscfg/internal/model"
+	rctypes "github.com/metal-toolbox/rivets/condition"
+)
+
+func (th *TaskHandler) HandleAction(ctx context.Context) error {
+	switch th.task.Parameters.Action {
+	case rctypes.ResetConfig:
+		return th.ResetBiosConfig(ctx)
+	case rctypes.SetConfig:
+		return th.SetBiosConfig(ctx)
+	default:
+		return th.failedWithError(ctx, string(th.task.Parameters.Action), errUnsupportedAction)
+	}
+}
+
+// ResetBios reset the bios of the server
+func (th *TaskHandler) ResetBiosConfig(ctx context.Context) error {
```

Comment:
The function name differs from the doc comment. Also, maybe we should make this private and have HandleAction be the public facing method for running tasks?